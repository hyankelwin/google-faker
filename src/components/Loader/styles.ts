import styled from 'styled-components';
import 'react-loading-skeleton/dist/skeleton.css';

export const Container = styled.div`
  padding-top: 20px;
  padding-bottom: 10px;
`;
