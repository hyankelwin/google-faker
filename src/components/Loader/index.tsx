import React from 'react';
import Skeleton from 'react-loading-skeleton';
import { Container } from './styles';

const Loader: React.FC = () => (
  <>
    <Container>
      <Skeleton count={1} height={20} width="30%" />
      <Skeleton count={1} height={30} width="60%" />
      <Skeleton count={1} height={20} width="90%" />
    </Container>
    <Container>
      <Skeleton count={1} height={20} width="30%" />
      <Skeleton count={1} height={30} width="60%" />
      <Skeleton count={1} height={20} width="90%" />
    </Container>
    <Container>
      <Skeleton count={1} height={20} width="30%" />
      <Skeleton count={1} height={30} width="60%" />
      <Skeleton count={1} height={20} width="90%" />
    </Container>
    <Container>
      <Skeleton count={1} height={20} width="30%" />
      <Skeleton count={1} height={30} width="60%" />
      <Skeleton count={1} height={20} width="90%" />
    </Container>
    <Container>
      <Skeleton count={1} height={20} width="30%" />
      <Skeleton count={1} height={30} width="60%" />
      <Skeleton count={1} height={20} width="90%" />
    </Container>
    <Container>
      <Skeleton count={1} height={20} width="30%" />
      <Skeleton count={1} height={30} width="60%" />
      <Skeleton count={1} height={20} width="90%" />
    </Container>
  </>
);

export default Loader;
