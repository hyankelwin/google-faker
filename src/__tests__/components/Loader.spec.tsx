import { mount } from 'enzyme';
import Loader from '../../components/Loader';

jest.mock(
  '../../../node_modules/react-loading-skeleton/dist/skeleton.css',
  () => {
    return {};
  },
);

describe('Loader component', () => {
  describe('should be able to passed in all tests on index.tsx', () => {
    it('should be able to render Loader component with success', () => {
      const wrapper = mount(<Loader />);

      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
