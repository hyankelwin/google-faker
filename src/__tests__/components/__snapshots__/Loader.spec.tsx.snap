// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should be able to passed in all tests on index.tsx should be able to render Loader component with success 1`] = `
"<Loader>
  <styled.div>
    <div className=\\"sc-aXZVg gJdRzT\\">
      <Skeleton count={1} height={20} width=\\"30%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={30} width=\\"60%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={20} width=\\"90%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
    </div>
  </styled.div>
  <styled.div>
    <div className=\\"sc-aXZVg gJdRzT\\">
      <Skeleton count={1} height={20} width=\\"30%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={30} width=\\"60%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={20} width=\\"90%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
    </div>
  </styled.div>
  <styled.div>
    <div className=\\"sc-aXZVg gJdRzT\\">
      <Skeleton count={1} height={20} width=\\"30%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={30} width=\\"60%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={20} width=\\"90%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
    </div>
  </styled.div>
  <styled.div>
    <div className=\\"sc-aXZVg gJdRzT\\">
      <Skeleton count={1} height={20} width=\\"30%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={30} width=\\"60%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={20} width=\\"90%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
    </div>
  </styled.div>
  <styled.div>
    <div className=\\"sc-aXZVg gJdRzT\\">
      <Skeleton count={1} height={20} width=\\"30%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={30} width=\\"60%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={20} width=\\"90%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
    </div>
  </styled.div>
  <styled.div>
    <div className=\\"sc-aXZVg gJdRzT\\">
      <Skeleton count={1} height={20} width=\\"30%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={30} width=\\"60%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
      <Skeleton count={1} height={20} width=\\"90%\\">
        <span className={[undefined]} data-testid={[undefined]} aria-live=\\"polite\\" aria-busy={true}>
          <span className=\\"react-loading-skeleton\\" style={{...}}>
            ‌
          </span>
          <br />
        </span>
      </Skeleton>
    </div>
  </styled.div>
</Loader>"
`;
